builder/api/pom.xml

```diff
@@ -38,11 +39,8 @@
         </dependency>
         <dependency>
             <groupId>io.helidon.common</groupId>
-            <artifactId>helidon-common-types</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>io.helidon.builder</groupId>
-            <artifactId>helidon-builder</artifactId>
+            <artifactId>helidon-common-config</artifactId>
+            <optional>true</optional>
```

Comment:
No it is not. Config is only required when the blueprint is annotated with `@Configured`.
In all other cases it will not be used in the generated code.
If you annotate it with `@Configured`, your module should have its own dependency on config already, as you are probably going to use the `config` method on the builders.

Reply:
Do you have a test module that can assure those dependencies don't leak in?  It would be definitely worth having that. ie., a test module that has test cases where none of them are configurable, and add a comment in the pom.xml to never allow config dependencies, etc.

Reply:
a simple PackagingTest would do it - from test scope, get the UrlClassLoader of the the test class, eumerate the artifacts, and ensure no helidon-common-modules in the list.

Reply:
Added encforcer plugin configuration to make sure this module never requires config